### PR TITLE
test: prometheus scale down e2e test

### PR DESCRIPTION
**Pull request by lihongyan1**
This should fix test failure [COO-79](https://issues.redhat.com/browse/COO-79)
`monitoring_stack_controller_test.go:585: assertion failed: 0 (prom.Status.Replicas int32) != 1 (int32)`
Also fix
[COO-84](https://issues.redhat.com/browse/COO-84) [QE] Case Alertmanager_runs_in_HA_mode is not stable
`monitoring_stack_controller_test.go:99: statefulsets.apps "alertmanager-alerting" not found`

### CODE REVIEW

**Comment by lihongyan1** [comment]
@simonpasquier @jan--f Please help review when convenient

File: test/e2e/monitoring_stack_controller_test.go [hunk 1]
```diff
@@ -577,29 +578,11 @@ func prometheusScaleDown(t *testing.T) {
...
-		if prom.Status.Replicas != 0 {
-			return false, nil
-		}
-		return true, nil
-	})
-
-	if wait.Interrupted(err) {
-		t.Fatal(fmt.Errorf("Prometheus was not scaled down"))
-	}
+	f.AssertPrometheusReplicaStatus(ms.Name, ms.Namespace, numOfRep)
```
Comment (jan--f) [inline test/e2e/monitoring_stack_controller_test.go:585]:
```suggestion
	f.AssertPrometheusReplicaStatus(ms.Name, ms.Namespace, 0)
```
Should this test for scale down to 0?

File: test/e2e/framework/assertions.go [hunk 2]
```diff
@@ -335,3 +335,33 @@ func getConditionByType(conditions []v1alpha1.Condition, ctype v1alpha1.Conditio
...
+			if err := f.K8sClient.Get(context.Background(), key, &prom); errors.IsNotFound(err) {
+				return false, nil
+			}
+			if prom.Status.Replicas != expectedReplicas {
+				return false, nil
+			}
+			return true, nil
+
+		}); wait.Interrupted(err) {
+			t.Fatal(fmt.Errorf("Prometheus %s/%s was not scaled down to %d", namespace, name, expectedReplicas))
```
Comment (jan--f) [inline test/e2e/framework/assertions.go:364]:
```suggestion
			t.Fatal(fmt.Errorf("Prometheus %s/%s has unexpected number of replicas, got %d, expected %d", namespace, name, prom.Status.Replicas, expectedReplicas))
```

**Review by jan--f — CHANGES_REQUESTED** [review]

File: test/e2e/monitoring_stack_controller_test.go (hunk 1 above)
Comment (lihongyan1) [reply test/e2e/monitoring_stack_controller_test.go:585]: Sorry typo, thanks your sharp eyes

**Comment by lihongyan1** [comment]
The fix of PR https://github.com/rhobs/observability-operator/pull/454 can work, case Alertmanager_runs_in_HA_mode failed in https://github.com/rhobs/observability-operator/actions/runs/8660157870/job/23747502573?pr=448, will provide new update to the case Alertmanager_runs_in_HA_mode

**Review by jan--f — APPROVED** [review]
/lgtm

**Pull request merged**